[DO NOT MERGE] Release 2.18 - Migration Customizations

djtaylor opened the pull request:
This is a PR to track the changes made to the `2.18` release branch to support converting our existing clusters to Kubespray.

djtaylor committed:
Disable swap task in check mode

djtaylor committed:
Properly disable swap in check mode

djtaylor committed:
Disable all swap tasks on check mode

djtaylor committed:
Disable apt cache update on check mode

djtaylor committed:
Get cgroup driver for docker even in check mode

djtaylor committed:
Add containerd 1.6.9 as an option

djtaylor committed:
Enable logging for download file

djtaylor committed:
Allow kubectl container check to run always

djtaylor committed:
check mode behavior

djtaylor committed:
Delegate calico version lookup

djtaylor committed:
ok to create download dir in check mode

djtaylor committed:
Check mode indentation

djtaylor committed:
Skip copy commands in dl when check

djtaylor committed:
get netfilter modinfo even in check mode

djtaylor committed:
Let it check kernel params in check mode

djtaylor committed:
Ok to generate token dir in check mode

djtaylor committed:
Ok to gather kube facts in check mode

djtaylor committed:
Skip copy kubectl dl when all dls skipped

djtaylor committed:
calc etcd cert serial in check mode

djtaylor committed:
make sure etcdctl version check runs api v3

djtaylor committed:
Enable checks in check mode

djtaylor committed:
Fix check mode syntax

djtaylor committed:
allow kubeadm token gen and check CA cert hash in check mode

djtaylor committed:
Check api when doing node labels even in check mode

djtaylor committed:
Check calico ip pool in check mode

djtaylor committed:
Calico check mode and account for skipped downloads

djtaylor committed:
Syntax fix

djtaylor committed:
Allow check kube proxy ds in check mode

djtaylor committed:
apps check mode tasks

djtaylor committed:
Remove async mode in favor of single node delegation

djtaylor committed:
Only try to gather tokens if any present

djtaylor committed:
Add options for webhook auth cache params

djtaylor committed:
Support extra args for kube proxy

djtaylor committed:
Allow containerd 1.6.8

djtaylor committed:
Ignore errors in token retrieval in check mode

djtaylor committed:
Ignore errors in etcdctl copy when in check mode

djtaylor committed:
Remove docker daemon.json from legacy installs

djtaylor committed:
Add cmd to gracefully convert an existing cluster

djtaylor committed:
Move task to ensure containerd version prior to core playbook tasks

djtaylor committed:
Don't dupe containerd package name

djtaylor committed:
install containerd after adding docker ce repo

djtaylor committed:
Run final task in serial

djtaylor committed:
Add a workaround for missing kube-dns svc

djtaylor committed:
Add template for placeholder kube-dns svc

djtaylor committed:
Delete the coredns configmap prior to upgrade to fix errors

djtaylor committed:
Add a workaround for recreating coredns cm

djtaylor committed:
Remove env from workaround play

djtaylor committed:
State to present

djtaylor committed:
svc to cm

djtaylor committed:
rename resources

djtaylor committed:
Add option to specify controller manager port

djtaylor committed:
Dont need the port argument

djtaylor committed:
Change serial to 1 globally

djtaylor committed:
Wait for 5m after finishing node before uncordon

djtaylor committed:
temporary disable of already run plays

djtaylor committed:
Remove temp disabled plays

djtaylor committed:
Check api status even in check mode

djtaylor committed:
Increase wait time to 20m

djtaylor committed:
Comment out tasks that have already run

djtaylor committed:
uncomment plays

djtaylor committed:
Workers in serial 2

djtaylor committed:
Disable plays while we troubleshoot worker node issues

djtaylor committed:
Enable last tasks

djtaylor committed:
enable all plays again

djtaylor committed:
Disable prefer_udp, set ttl 30

benlangfeld commented:
Is this intended to form a todo list of things we'll shoot to align with upstream after we're using kubespray consistently, or do we expect to contribute these changes upstream?

djtaylor committed:
Revert ttl change

djtaylor committed:
Ability to template dns/resolv options

djtaylor committed:
Set serial back to 1

mvleandro committed:
Configuring repositoty to be used as an Ansible collection.

mvleandro committed:
Merge pull request #4 from powerhome/feat/collection

Configuring repositoty to be used as an Ansible collection.

mvleandro committed:
Fixing Ansible collection repository address and authors. (#5)

mvleandro committed:
Changing file extension.

mvleandro committed:
Just a test.

mvleandro committed:
Update main.yml

mvleandro committed:
Update galaxy.yml

Changing version to a compliant git version pattern.

mvleandro committed:
Update galaxy.yml

mvleandro committed:
Update galaxy.yml

mvleandro committed:
Update galaxy.yml

mvleandro committed:
Update galaxy.yml

mvleandro committed:
Adding invoked playbooks inside the playbooks directory.

mvleandro committed:
Making it usable as a collection.

Moumoutaru committed:
Skip task when in check mode

- Skip arbitrary wait time when in check mode

In check mode there is no need for us to wait for 20m. When executing from command line this isn't an issue cause we can ctrl+c to cancel the wait but when runnning via tower we'd end up waiting 20m for no reason.

iblackman committed:
Add support to calico v3.22

iblackman committed:
calico crds download for version higher than v3.22.3

iblackman committed:
Merge pull request #6 from powerhome/feat/add-support-to-calico-3-22

Add support to calico v3.22

Moumoutaru committed:
Add checksums for version v1.21.14

- Adds kubectl, kubelet, and kubeadm sha256 checksums for the v1.21.14 binaries

Moumoutaru committed:
Merge pull request #7 from powerhome/add-checksums-for-v1.21.14

Add checksums for version v1.21.14

Moumoutaru committed:
test file for theory

- Test file for theory

Moumoutaru committed:
Remove test_file

- Removes test file for submodule tests

Moumoutaru committed:
Adding checksums for v1.22.17

- Adding checksums for v1.22.17

Moumoutaru committed:
Adjusting order

- Adjusting order

Moumoutaru committed:
Merge pull request #9 from powerhome/add-checksums-for-v1.22.17

Adding checksums for v1.22.17

bcdonadio committed:
Implement mechanism for DNS federation

bcdonadio committed:
Add new RBAC permissions for CoreDNS version

bcdonadio committed:
Fix case when federation is not enabled

bcdonadio committed:
Fix potential clash with non-svc subdomains

Commit:
Merge pull request #12 from powerhome/feat/dns-fed

Implement DNS federation mechanism

Moumoutaru committed:
cluster.yaml var inclusions

- Include vars for cluster.yaml

Moumoutaru committed:
Merge pull request #14 from powerhome/release-2.18-power-1-cluster

cluster.yaml var inclusions
